src/olympia/reviewers/tests/test_utils.py

```diff
@@ -1584,6 +1517,72 @@ def test_reject_multiple_versions(self):
         # Check points awarded.
         self._check_score(amo.REVIEWED_EXTENSION_MEDIUM_RISK)
 
+    def test_reject_multiple_versions_with_delay(self):
+        old_version = self.version
+        self.version = version_factory(addon=self.addon, version='3.0')
+        AutoApprovalSummary.objects.create(
+            version=self.version, verdict=amo.AUTO_APPROVED, weight=101)
+        # An extra file should not change anything.
+        file_factory(version=self.version, platform=amo.PLATFORM_LINUX.id)
+        self.setup_data(amo.STATUS_APPROVED, file_status=amo.STATUS_APPROVED)
+
+        in_the_future = datetime.now() + timedelta(days=14)
+
+        # Safeguards.
+        assert isinstance(self.helper.handler, ReviewFiles)
+        assert self.addon.status == amo.STATUS_APPROVED
+        assert self.file.status == amo.STATUS_APPROVED
+        assert self.addon.current_version.is_public()
+
+        data = self.get_data().copy()
+        data.update({
+            'versions': self.addon.versions.all(),
+            'delayed_rejection': True,
+            'delayed_rejection_days': 14,
+        })
+        self.helper.set_data(data)
+        self.helper.handler.reject_multiple_versions()
+
+        # File/addon status didn't change.
+        self.addon.reload()
+        self.file.reload()
+        assert self.addon.status == amo.STATUS_APPROVED
+        assert self.addon.current_version == self.version
+        assert list(self.addon.versions.all()) == [self.version, old_version]
+        assert self.file.status == amo.STATUS_APPROVED
+
+        # The versions are now pending rejection.
+        for version in self.addon.versions.all():
+            assert version.pending_rejection
+            self.assertCloseToNow(version.pending_rejection, now=in_the_future)
+
+        assert len(mail.outbox) == 1
+        assert mail.outbox[0].to == [self.addon.authors.all()[0].email]
```

Comment:
Nit: Would a `email = mail.outbox[0]` here make the following code a bit more succinct?

Reply:
There were quite a few tests to patch... but I think yeah, it's worth doing. 9343bbd66aeda3c992c09d7093b9eca89a041086